heat/core/linalg.py

```diff
@@ -145,7 +163,7 @@ def matmul(a, b):
         b = c_type(b, device=b.device)
 
     if a.split is None and b.split is None:  # matmul from torch
-        if len(a.gshape) < 2 or len(b.gshape) < 2:
+        if len(a.gshape) < 2 or len(b.gshape) < 2 or no_resplit:
             # if either of A or B is a vector
             return factories.array(
                 torch.matmul(a._DNDarray__array, b._DNDarray__array), device=a.device
```

Comment:
two lines below (ll 172) a = a.resplit_(0) is not the appropriate syntax: resplit_ is inplace, but this gives the impression that it is out of place. 
Better would be a.resplit_(0)